clrenv/lazy_env.py

```diff
@@ -40,18 +43,35 @@ def set_mode(self, *mode):
     def get_mode(self):
         return self.__mode
 
+    def clear_runtime_overrides(self):
+        self.__runtime_overrides.clear()
+
     def __getattr__(self, key):
         if self.__env is None:
             self.__env = get_env(*self.__mode)
 
-        try:
-            return getattr(self.__env, key)
-        except AttributeError:
-            return None
+        if key in self.__runtime_overrides:
+            return self.__runtime_overrides[key]
+        return getattr(self.__env, key, None)
 
     def __getitem__(self, key):
         return self.__getattr__(key)
 
+    def __setattr__(self, key, value):
+        # Internal fields are prefixed with a _
+        if key.startswith('_'):
+            return object.__setattr__(self, key, value)
+
+        # Ideally we wouldn't be overriding global state like this at all, but at least
+        # make it loud.
+        print(f'Manually overriding env.{key} to {value}.', file=sys.stderr)
```

Comment:
nit: Should `logger` be printing this statement instead of just `print`?

Reply:
good call